### ISSUE TITLE
508-defect-2: Hint text not read out

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#54215]
### Point of contact

Josh Kim

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

[Label descriptions should be styled as hint text to match design system guidelines and be consistent with other pages in the flow.](https://github.com/department-of-veterans-affairs/va.gov-team/issues/54215#issuecomment-1451937977)

### Reproduction steps

1. Log in to any user
2. Start [Supplemental Claim form](https://staging.va.gov/decision-reviews/supplemental-claim/file-supplemental-claim-form-20-0995)
3. Get to contestable issues page
4. Use "Add a new issue" link
5. Start screen reader and note that the sentence below the hint ("Enter the date on your decision notice (the letter you received in the mail from us)." is not read out with the label

#### List of hint text with this issue
- [ ] What is your primary phone "We may need to contact you if we have questions"
- [ ] Add a new issue "Enter the date on your decision notice..."

### Proposed solution or next steps

<img width="1486" alt="Screenshot 2023-03-02 at 9 05 43 AM" src="https://user-images.githubusercontent.com/14154792/222452075-b29ce46f-671b-4ef7-bdba-8200ffb1422c.png">

Gave this a double check and went into the code for a bit! This looks to be more of a DST issue than anything, specifically:
- the hint prop for input fields isn't programmatically linked to the input itself (credit to @jaredcunha for catching this)
- we need to determine how we code hint text to work for both the input field and the memorable date; [Jared documents two possible solutions here](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1334#issuecomment-1322553465) I lean towards the former as it's the more semantic pick (hint text is _describing_ the field, not technically a part of the label itself)
 - memorable date does the latter 😓 which results in a really verbose experience when we include instructions on how to use the memorable date 

To fix these, I propose the following:
- [ ] DST fixes the hint prop for input fields to use @jaredcunha 's `aria-describedby` recommendation
- [ ] DST adjusts the hint prop for memorable date fields to also use @jaredcunha `aria-describedby` recommendation (and not add directly within the `legend`)
- [ ] We don't immediately show the formatting description for the memorable date field, but provide it as part of error validation

### References, articles, or WCAG support

1. [Design system hint text guidance](https://design.va.gov/components/form/#default-hint-text)
2. [Using `aria-describedby`](https://www.w3.org/WAI/tutorials/forms/instructions/#using-aria-describedby)

### Type of issue

- [X] Screenreader
- [ ] Keyboard
- [ ] Focus
- [ ] Headings
- [ ] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [ ] Something else

**Comment by Mottie** [comment]
Ready for review on staging!

**Comment by joshkimux** [comment]
<img width="1486" alt="Screenshot 2023-03-02 at 9 05 43 AM" src="https://user-images.githubusercontent.com/14154792/222452075-b29ce46f-671b-4ef7-bdba-8200ffb1422c.png">

Gave this a double check and went into the code for a bit! This looks to be more of a DST issue than anything, specifically:
- the hint prop for input fields isn't programmatically linked to the input itself (credit to @jaredcunha for catching this)
- we need to determine how we code hint text to work for both the input field and the memorable date; [Jared documents two possible solutions here](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1334#issuecomment-1322553465) I lean towards the former as it's the more semantic pick (hint text is _describing_ the field, not technically a part of the label itself)
 - memorable date does the latter 😓 which results in a really verbose experience when we include instructions on how to use the memorable date 

To fix these, I propose the following:
- [ ] DST fixes the hint prop for input fields to use @jaredcunha 's `aria-describedby` recommendation
- [ ] DST adjusts the hint prop for memorable date fields to also use @jaredcunha `aria-describedby` recommendation (and not add directly within the `legend`)
- [ ] We don't immediately show the formatting description for the memorable date field, but provide it as part of error validation

**Comment by saderagsdale** [comment]
Removing this from the sprint since it relies on changes to the design system @Mottie @joshkimux. Let me know if you see alternative paths forward.

**Comment by joshkimux** [comment]
@saderagsdale thanks Sade! I don't see any alternative paths (that would be worth the effort) forward in the meantime until the DST can resolve these open issues.

**Comment by joshkimux** [comment]
@Mottie not much we can do, I'll mark this as blocked!

**Comment by joshkimux** [comment]
Now marked as blocked via label and pipeline!

**Comment by saderagsdale** [comment]
@joshkimux Moving this to the product backlog until it's actionable.

**Comment by davidakennedy** [comment]
<img width="726" alt="Contestable issues page showing an arrow pointing to the div containing the Enter the date on your decision notice (the letter you received in the mail from us) inside a legend element." src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/1473618/5c2b225c-61c5-40e4-9dde-ca0996826bf3">

<img width="561" alt="What is your primary phone number? page with an arrow pointing to the div containing the text and the text on the screen." src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/1473618/1390829f-2a07-4ac3-b779-de6ae717ce19">

I did some testing on this, to see what's changed, of anything. A few updates:

- The markup here has changed, thanks to the move to the v3 components. The hint text is now inside a legend, which allows it to be read out by screen readers and other assistive technology. This is confirmed by the two screenshots added.
- This approach is actually better than using `aria-descibedby` because it will always get read out by assistive technology, whereas the `aria-descibedby` may not in some situations.
- I don't think we should change the color of the hint text unless we approach doing that at the design system level.

Given all this, we should be able to close this because the primary issues are fixed. I'll mark it as closed, but @saderagsdale and @Mottie, shout if you think otherwise.
